jax/interpreters/partial_eval.py

```diff
@@ -2156,24 +2159,23 @@ def _collect_implicit(
   idxs: Dict[AbstractedAxisName, DBIdx] = {}
   implicit_types: List[AbstractValue] = []
   explicit_tracers: Dict[TracerId, int] = {}
-  counter = (DBIdx(i) for i in it.count())
+  counter = it.count()
 
   # Add implicit arguments to idxs.
-
   for explicit_idx, (x, spec) in enumerate(zip(args, specs)):
     for i, name in spec.items():
       if name not in idxs and id(x.shape[i]) not in explicit_tracers:
-        idxs[name] = next(counter)
+        idxs[name] = DBIdx(next(counter))
         implicit_types.append(raise_to_shaped(get_aval(x.shape[i])))
     if isinstance(x, Tracer):
-      explicit_tracers[id(x)] = explicit_idx
+      explicit_tracers.setdefault(id(x), explicit_idx)  # use the first
```

Comment:
Should we add a test for this as well?

Reply:
That's it, I'm removing you as coauthor.